### ISSUE TITLE
Duplicate content-type headers set in OpenSearch python client  

### ISSUE THREAD

**Issue by samodelkinas** [aws/aws-sdk-pandas#1965]
OpenSearch write fails with duplicate content-type headers due to https://github.com/aws/aws-sdk-pandas/blob/52fe27f332e1f475b71fba669af7e708074f1970/awswrangler/opensearch/_write.py#L131
When client is created manually (for on-premises opensearch, using https://github.com/opensearch-project/opensearch-py/blob/58aa041db325cd9c9d273f21c00c969f272e933c/opensearchpy/connection/base.py#L105), duplicate headers are created in lowercase. 

```
client = op.OpenSearch(
        hosts=[{'host': config('OPENSEARCH_INSTANCE_PRIMARY'), 'port': 9200}],
        http_compress=True,
        http_auth=(config('OPENSEARCH_USERNAME'),
                   config('OPENSEARCH_PASSWORD')),
        use_ssl=True,
        verify_certs=False,
        ssl_assert_hostname=False,
        ssl_show_warn=False,
        timeout=60
    )
wr.opensearch.index_df(
        client=client,
        df=nr_api.entities,
        index=samples_entities_index,
        bulk_size=10000
    )
```

This code would produce duplicate content-type headers error

this also affects _read.py for OpenSearch SQL queries. In both cases issue can be resolved by renaming headers to lowercase in _write.py and _read.py, or removing them completely. Ideally, this should be addressed in OpenSearch python client itself, i.e. implement case insensitive check for existing content-type header

**Comment by jaidisido** [comment]
If I understand correctly you are using an OpenSearch [connection](https://opensearch-project.github.io/opensearch-py/api-ref/connection.html) whereas our [method](https://github.com/aws/aws-sdk-pandas/blob/main/awswrangler/opensearch/_utils.py#L226) is using the [client](https://opensearch-project.github.io/opensearch-py/api-ref/clients/opensearch_client.html). Out of curiosity, any reason why you prefer the former when the client supports the same arguments?

Happy to make `Content-Type` lowercase in any case.

**Comment by samodelkinas** [comment]
Hey. Thanks a lot for getting back on this. I placed the actual code snippet within edit couple of days ago, We are using client to connect to on premises OpenSearch, not the connection, and it fails with any operation that requires content-type headers. That includes writing data frame to index or querying using SQL dialect, so an update needs to be made in both _write.py and _read.py. I have monkeypatched those 2 files so far to go on  with further testing.
